upgrade50Plus.sh

```diff
@@ -55,13 +59,62 @@ print_dd_progress() {
     done    
 } 
 
-
 check_preconditions() {
+    pretty "Starting upgrade..." "" "" "Starting upgrade..." ""
+    sleep 5
+    pretty "Do NOT turn off" "till complete..." "" "Do NOT turn off" "till complete..."
+    sleep 5
+
     pretty "Checking preconditions..." "" "" "Checking" "preconditions..."
     [ -z "$IP" ] && quit "Usage:" "$0 <IP-of-ePC>" "" "wrong usage" ""
+    rm /root/.ssh/known_hosts &> /dev/null
     ping -c1 $IP 1>&2 > /dev/null || quit "No connection!" "ePC is not reachable at" "$IP, update failed." "Update failed." "ePC not reachable."
-    executeAsRoot "sfdisk -d /dev/sda | grep sda5 | grep 42049536" || quit "Unexpected partition" "ePC partition 5 is not at" "expected position, update failed." "Update failed." "partition error"
-    executeAsRoot "sfdisk -d /dev/sda | grep sda5 | grep 20482422" || quit "Unexpected partition" "ePC partition 5 is not of" "expected size, update failed." "Update failed." "partition error"
+    sshpass -p 'TEST' ssh -o StrictHostKeyChecking=no TEST@$IP "exit" &> /dev/null || quit "Can't login into Windows" "Update failed." "" "Update failed." "no win login"
+
+    #get SSD_SIZE of the ePC
+    SSD_SIZE=$(sshpass -p 'TEST' ssh -o StrictHostKeyChecking=no TEST@$IP "wmic diskdrive get size")
+    SSD_SIZE=$(echo "$SSD_SIZE" | sed -n 2p)        # Size Info is in the second line
+    SSD_SIZE=${SSD_SIZE//[ $'\001'-$'\037']}        # remove possible DOS carriage return characters
+    SSD_SIZE=$((SSD_SIZE / 1000000000))
+
+    # switch from Windows to Ubuntu
```

Comment:
This should be extracted into a separate function, for example as the comment states: "switchFromWindowsToUbuntu()"

Reply:
Also, switching from windows to ubuntu is not exactly "checkingPreconditions". If somehow possible, the checking should not change any state and let the system untouched. If - in the proposed implementation - the SSD_SIZE check fails, the system is unusable for the user, as ubuntu without audio-engine remains the active boot partition.